stylize.py

```diff
@@ -96,7 +96,12 @@ def stylize(network, initial, content, styles, iterations,
         loss = content_loss + style_loss + tv_loss
 
         # optimizer setup
-        train_step = tf.train.AdamOptimizer(learning_rate).minimize(loss)
+
+        def train_step(session = None ,iterations = 0):
+            if optimizer == 'adam':
+                tf.train.AdamOptimizer(learning_rate).minimize(loss)
+            elif optimizer == 'lbfgs':
+                tf.contrib.opt.ScipyOptimizerInterface(loss, method='L-BFGS-B', options={'maxiter': iterations}).minimize(session)
```

Comment:
About `options={'maxiter': iterations}`:
Setting `options={'maxiter': 1}` here and run the iterations in the "adam way" will run out of memory in the first 100 iterations on my GTX970. But run all the iterations in 1 time works fine. I guess somewhere of it is leaking memory.